src/compression/snappy.rs

```diff
@@ -1,18 +1,37 @@
 use std::io::{self, Read};
 
 use byteorder::{BigEndian, ByteOrder};
-use snappy;
+use snap;
 
 use error::{Result, Error};
 
 pub fn compress(src: &[u8]) -> Result<Vec<u8>> {
-    Ok(snappy::compress(src))
+    let mut buf = vec![0; snap::max_compress_len(src.len())];
+
+    snap::Encoder::new()
+        .compress(src, &mut buf)
+        .map(|len| {
+                 buf.truncate(len);
+
+                 buf
+             })
+        .map_err(|err| Error::InvalidSnappy(err))
 }
 
 fn uncompress_to(src: &[u8], dst: &mut Vec<u8>) -> Result<()> {
-    snappy::uncompress_to(src, dst)
-        .map(|_| ())
-        .map_err(|_| Error::InvalidInputSnappy)
+    snap::decompress_len(src)
+        .and_then(|min_len| {
+            let off = dst.len();
```

Comment:
* we should handle `min_len == 0` here; so far i can see `snap::decompress_len` return it if `src.is_empty()`.
* btw: it's unfortunate we have to parse the header of the input twice, once as part of the call to `decompress_len` and once as part of `Decoder::new().decompress` ... can you think of a way to avoid this, and maybe suggest an API to `rust-snappy`/@BurntSushi which could avoid this overhead?